s2s.go

```diff
@@ -14,23 +14,35 @@ import (
 )
 
 type S2SClientConfig struct {
-	Service       string
-	JWTSecret     string
-	AccessKey     string
+	// JWTToken is the JWT token to use for authentication.
+	JWTToken string
+	// JWTSecret is the secret key used to create the JWT token.
+	JWTSecret string
+	// Service is used in the service claim of the JWT token.
+	Service string
+	// AccessKey is an optional access key to use for authentication.
+	AccessKey string
+	// DebugRequests enables logging of HTTP requests.
 	DebugRequests bool
 }
 
 // Service-to-service HTTP client for internal communication between Sequence services.
+// If JWTSecret is provided, it will create a HS256 JWT token with the service name in the claims.
+// If both JWTSecret and JWTToken are provided, JWTToken will take precedence.
 func S2SClient(cfg *S2SClientConfig) *http.Client {
 	serviceName := cmp.Or(cfg.Service, filepath.Base(os.Args[0]))
 
 	httpClient := &http.Client{
 		Transport: transport.Chain(http.DefaultTransport,
 			traceid.Transport,
 			transport.SetHeader("User-Agent", fmt.Sprintf("sequence/%s", serviceName)),
-			transport.If(cfg.JWTSecret != "",
+			transport.If(cfg.JWTSecret != "" || cfg.JWTToken != "",
```

Comment:
For the static JWTToken, we don't need `SetHeaderFunc()` (which is invoked on each request) - we can use `SetHeader()`, which is meant for static creds.

Let me handle this.